code/game/objects/items/pillow.dm

```diff
@@ -0,0 +1,154 @@
+
+/obj/item/pillow
+	name = "pillow"
+	desc = "A soft and fluffy pillow. You can smack someone with this!"
+	icon = 'icons/obj/pillow.dmi'
+	icon_state = "pillow_1_t"
+	inhand_icon_state = "pillow_t"
+	lefthand_file = 'icons/mob/inhands/items/pillow_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/items/pillow_righthand.dmi'
+	w_class = WEIGHT_CLASS_NORMAL
+	damtype = STAMINA
+	var/last_fighter //whoever last use this pillow
+	var/obj/item/clothing/neck/pillow_tag/pillow_trophy
+	var/static/tag_desc = "This one seems to have its tag removed."
+	var/variation = 1 //for selecting the various sprite variation, defaults to the blank white pillow
+
+
+/obj/item/pillow/Initialize(mapload)
+	. = ..()
+	if(!pillow_trophy)
+		pillow_trophy = new(src)
+	AddComponent(/datum/component/two_handed, \
+		force_unwielded = 5, \
+		force_wielded = 10, \
+	)
+
+/obj/item/pillow/Destroy(force)
+	. = ..()
+	pillow_trophy = null
+
+/obj/item/pillow/attack(mob/living/carbon/target_mob, mob/living/user, params)
+	. = ..()
+	if(!iscarbon(target_mob))
+		return
+	if(HAS_TRAIT(src, TRAIT_WIELDED))
+		user.apply_damage(10, STAMINA) // when hitting with such force we should prolly be getting tired too
+	last_fighter = user
+	playsound(user, 'sound/items/pillow_hit.ogg', 80) //the basic 50 vol is barely audible
+
+/obj/item/pillow/attack_secondary(mob/living/carbon/victim, mob/living/user, params)
+	. = ..()
+	if(!isliving(victim))
+		return
+	if(victim.wear_mask)
+		return
+	if(!user.get_bodypart(BODY_ZONE_HEAD))
+		return
+	if(victim.body_position || user.grab_state >= GRAB_AGGRESSIVE)
+		to_chat(user, span_alert("You being to smother [victim]"))
+		to_chat(victim, span_alertwarning("[user] is begingin to smother you!"))
+		smothering(user, victim)
+
+/obj/item/pillow/proc/smothering(mob/living/carbon/user, mob/living/carbon/victim)
+	while(victim)
+		if(victim.body_position == FALSE && user.grab_state <= GRAB_NECK)
+			break
+		if(!do_after(user, 1 SECONDS, victim))
+			break
+		victim.losebreath += 1
+	victim.losebreath = 0
+	to_chat(victim, span_notice("You break out!"))
+	to_chat(user, span_notice("You stop smothering!"))
+
+/obj/item/pillow/examine(mob/user)
+	. = ..()
+	. += span_notice("<i>There's more information below, you can look again to take a closer look...</i>")
+
+/obj/item/pillow/examine_more(mob/user)
+	. = ..()
+	. += span_notice("Alt-click to remove the tag!")
+
+/obj/item/pillow/AltClick(mob/user)
+	. = ..()
+	if(!pillow_trophy)
+		balloon_alert(user, span_notice("no tag!"))
+		return
+	balloon_alert(user, span_notice("removing tag..."))
+	if(!do_after(user, 2 SECONDS, src))
+		return
+	if(last_fighter)
+		pillow_trophy.desc = "a pillow tag taken from [last_fighter] after a gruesome pillow fight."
+	user.put_in_hands(pillow_trophy)
```

Comment:
you should null out pillow_trophy BEFORE you call update_state, preferably right under here when you put it in hand.